### PR TITLE
Allow to change font of text elements

### PR DESCRIPTION
**Pull request by Feodor0090**
While all the HUD use "numeric" font as in triangles skin, text and information elements use torus font, as in argon skin. This doesn't let all the hud look in the same manner.
This PR allows user to choose the used font.

If the direction will be accepted, i will do the same with existing hud elements to allow making them argon-styled before new designs will be implemented.

### CODE REVIEW

**Comment by bdach** [comment]
I was worried this sort of pull was going to appear after the text elements were added. To me this is going down a bad path of scope creep. Next thing you know, users are going to demand to be able to choose arbitrary fonts from their system rather than pick from the limited in-game selection of fonts.

To me this text customisation stuff is a rabbit hole and doesn't belong in the game. But I've been overruled on that once before, so I'm going to state my opposition and leave it at that.

File: osu.Game/Skinning/DefaultFont.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    /// <summary>
+    /// The type of built-in font to use for <see cref="DefaultTextSkinComponent"/>.
+    /// </summary>
+    public enum DefaultFont
+    {
+        Venera,
+        Torus,
+        TorusAlt,
+        Inter
+    }
```
Comment (bdach) [inline osu.Game/Skinning/DefaultFont.cs:17]: - Why does this enum even exist if `Typeface` exists? Would get rid of the necessity to map the enums across and whatever.
- (The following applies to whichever enum survives) Can these have `[Description]`s or something? The dropdown at present looks really quite ugly what with the enum names printed directly to the user. Almost looks like a debug option or something.
  
  image

File: osu.Game/Skinning/Components/DefaultTextSkinComponent.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+using osu.Framework.Graphics.Sprites;
+using osu.Game.Configuration;
+using osu.Game.Graphics;
+
+namespace osu.Game.Skinning.Components
+{
+    /// <summary>
+    /// Skin element that contains text and have ability to control its font.
+    /// </summary>
+    public abstract partial class DefaultTextSkinComponent : Container, ISkinnableDrawable
```
Comment (bdach) [inline osu.Game/Skinning/Components/DefaultTextSkinComponent.cs:15]: What's the `Default` for in the name of this class? I don't quite get it.

**Review by bdach — COMMENTED** [review]
More prosaically:

File: osu.Game/Skinning/DefaultFont.cs (hunk 1 above)
Comment (Feodor0090) [reply osu.Game/Skinning/DefaultFont.cs:17]: Somehow missed the fact that it already existed. The only issue is that venera requires non-default font weight, but it still looks cleaner than dublicated enum.

File: osu.Game/Skinning/Components/DefaultTextSkinComponent.cs (hunk 2 above)
Comment (Feodor0090) [reply osu.Game/Skinning/Components/DefaultTextSkinComponent.cs:15]: Wanted to distinguish it from components based on "legacy" font, but I agree that it doesn't sound well.

Comment (peppy) [reply osu.Game/Skinning/Components/DefaultTextSkinComponent.cs:15]: I've renamed this.

**Comment by peppy** [comment]
Doing this using inheritance seems okay for now, but in the future, we may have more complex component structures and be forced to move this to an interface. We can deal with that when it happens (should not affect serialisation, which is why I'm okay with going with this simple solution for now).

> If the direction will be accepted, i will do the same with existing hud elements to allow making them argon-styled before new designs will be implemented.

This kind of use case may precisely hit the snag I mentioned here, though. If so, please consider switching to an interface implementation (with each usage doing its own `BindValueChanged` and moving the `FontUsage` lookup to a static helper methods or extension method).

**Review by peppy — APPROVED** [review]

**Pull request merged**